flybywiresim/a32nx
# fix(chrono): improved logic for operations

Fixes #5230 

## Summary of Changes

Improved overall logic for operation and display of the chronometer.

- Chrono computer works even when the aircraft is completely powered down (being connected to DC HOT BUS 1)
- Chrono display should work only when DC ESS is powered
- The buttons to control chrono and elapsed time work only when DC ESS is powered and become inactive once DC ESS is unpowered
- Chrono and Elapsed time counters once started, keep running, even if the display is off (i.e. connected to DC HOT BUS 1).

## Screenshots (if necessary)



## References

Komp




## Additional context

Thanks to @komp1821 for testing out the weird test cases on the real bus, and @davidwalschots for his constant guidance :)


Discord username (if different from GitHub): sidnov#8337

## Testing instructions


1. Spawn cold and dark, observe the chrono display is OFF
2. Turn on both batteries or external power (whichever available). Chrono display should turn on.
3. Test all the basic functions. Chrono, Elapsed Time, Time/Date.
4. Turn on the chronometer and elapsed time, then turn off the batteries/external power to put the aircraft in cold and dark state again. Observe chrono display turning off.
5. Try stopping the chronometer, elapsed time in the powered down state. Turn the batteries/external power on and observe chrono/ET is still running.
6. Refer to the summary of changes to find any edge cases.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
